pkg/progressreader/progressstatus.go

```diff
@@ -0,0 +1,72 @@
+package progressreader
+
+import (
+	"bytes"
+	"io"
+	"sync"
+
+	"github.com/docker/docker/vendor/src/github.com/Sirupsen/logrus"
+)
+
+type ProgressStatus struct {
```

Comment:
Please document all exported types according to [Go's Documentation Guidelines](http://blog.golang.org/godoc-documenting-go-code). We cannot be having a type like this without a single comment about what it is supposed to do.